src/screens/TopicPageScreen/index.js

```diff
@@ -59,7 +59,7 @@ const TopicPageScreen = (props) => {
   ).current;
 
   const [feedsContext, dispatch] = React.useContext(Context).feeds;
-  const feeds = feedsContext.topicFeeds;
+  const feeds = feedsContext.topicFeeds?.filter((feed) => feed?.topics?.includes(topicName));
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new filtering logic for `feeds` seems fine, but it might throw an error if `topicFeeds` is undefined. To prevent this, you should add a check to ensure `topicFeeds` is defined before calling the `filter` function.
```diff
- const feeds = feedsContext.topicFeeds?.filter((feed) => feed?.topics?.includes(topicName));
+ const feeds = feedsContext.topicFeeds ? feedsContext.topicFeeds.filter((feed) => feed?.topics?.includes(topicName)) : [];
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->